suites/src/main/java/com/hedera/block/suites/CustomThreadPoolExecutor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.suites;
+
+import static java.lang.System.Logger.Level.ERROR;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A custom implementation of {@link ThreadPoolExecutor} that provides specific configurations
+ * and enhanced error logging capabilities for tasks executed within the thread pool.
+ *
+ * <p>This executor is configured with:
+ * <ul>
+ *   <li>Core pool size: 8 threads
+ *   <li>Maximum pool size: 8 threads
+ *   <li>Keep-alive time: 10 seconds
+ *   <li>Work queue: {@link LinkedBlockingQueue} with default capacity
+ * </ul>
+ *
+ * <p>The executor overrides the {@link #afterExecute(Runnable, Throwable)} method to log any
+ * exceptions thrown by tasks after their execution, aiding in debugging and error tracking.
+ */
+public class CustomThreadPoolExecutor extends ThreadPoolExecutor {
```

Comment:
I suggest that we rename this to `SimpleErrorLoggingExecutor` or similar, since the name `Custom` is really not descriptive.

Reply:
Changed to `ErrorLoggingExecutor`